src/common/AdaptiveStream.h

```diff
@@ -89,50 +89,76 @@ class AdaptiveStream;
     bool StreamChanged() { return stream_changed_; }
 
   protected:
+    virtual bool parseIndexRange(PLAYLIST::CRepresentation* rep, const std::string& buffer);
+
+    virtual void SetLastUpdated(std::chrono::system_clock::time_point tm) {};
```

Comment:
```suggestion
    virtual void SetLastUpdated(const std::chrono::system_clock::time_point tm) {};
```

Reply:
this is not part of the PR change
this PR need a rebase with the linked PR that need to be merged before this
so you have to exclude the first commit from the review